src/expr/src/vector_op/string.rs

```diff
@@ -381,3 +381,87 @@ pub fn quote_ident(s: &str, writer: &mut dyn Write) {
     }
     write!(writer, "\"").unwrap();
 }
+
+/// Returns the first n characters in the string.
+/// If n is a negative value, the function will return an empty string.
+///
+/// # Example
+///
+/// ```slt
+/// query T
+/// select left('RisingWave', 6)
+/// ----
+/// Rising
+///
+/// query T
+/// select left('RisingWave', 0)
+/// ----
+///
+///
+/// query T
+/// select left('RisingWave', -1)
+/// ----
+#[function("left(varchar, int32) -> varchar")]
+pub fn left(s: &str, n: i32, writer: &mut dyn Write) {
+    s.chars()
+        .take(n.max(0) as usize)
+        .for_each(|c| writer.write_char(c).unwrap());
+}
+
+/// Returns the last n characters in the string.
+/// If n is a negative value, the function will return an empty string.
+///
+/// # Example
+///
+/// ```slt
+/// query T
+/// select right('RisingWave', 4)
```

Comment:
can we add a test case where the number is larger than the length of string input

Reply:
definitely, adding now

Reply:
Thanks!